python/treelite/sklearn/__init__.py

```diff
@@ -97,4 +106,131 @@ class SKLRFMultiClassifierConverter(SKLRFMultiClassifierMixin, SKLConverterBase)
     pass
 
 
-__all__ = ['import_model']
+class ArrayOfArrays:
+    """
+    Utility class to marshall a collection of arrays in order to pass to a C function
+    """
+    def __init__(self, *, dtype):
+        int64_ptr_type = ctypes.POINTER(ctypes.c_int64)
+        float64_ptr_type = ctypes.POINTER(ctypes.c_double)
+        if dtype == np.int64:
+            self.ptr_type = int64_ptr_type
+        elif dtype == np.float64:
+            self.ptr_type = float64_ptr_type
+        else:
+            raise ValueError(f'dtype {dtype} is not supported')
+        self.dtype = dtype
+        self.collection = []
+
+    def add(self, array, *, expected_shape=None):
+        """Add an array to the collection"""
+        assert array.dtype == self.dtype
+        if expected_shape:
+            assert array.shape == expected_shape, \
+                    f'Expected shape: {expected_shape}, Got shape {array.shape}'
+        v = np.array(array, copy=False, dtype=self.dtype, order='C')
+        self.collection.append(v.ctypes.data_as(self.ptr_type))
+
+    def as_c_array(self):
+        """Prepare the collection to pass as an argument of a C function"""
+        return c_array(self.ptr_type, self.collection)
+
+
+def import_model_v2(sklearn_model):
+    # pylint: disable=R0914,R0912
+    """
+    Load a tree ensemble model from a scikit-learn model object
+
+    Parameters
+    ----------
+    sklearn_model : object of type \
+                    :py:class:`~sklearn.ensemble.RandomForestRegressor` / \
+                    :py:class:`~sklearn.ensemble.RandomForestClassifier` / \
+                    :py:class:`~sklearn.ensemble.ExtraTreesRegressor` / \
+                    :py:class:`~sklearn.ensemble.ExtraTreesClassifier` / \
+                    :py:class:`~sklearn.ensemble.GradientBoostingRegressor` / \
+                    :py:class:`~sklearn.ensemble.GradientBoostingClassifier`
+        Python handle to scikit-learn model
+
+    Returns
+    -------
+    model : :py:class:`~treelite.Model` object
+        loaded model
+    """
+    class_name = sklearn_model.__class__.__name__
+    module_name = sklearn_model.__module__.split('.')[0]
+    if module_name != 'sklearn':
+        raise TreeliteError('Not a scikit-learn model')
+
+    if class_name in ['RandomForestRegressor', 'ExtraTreesRegressor', 'GradientBoostingRegressor',
```

Comment:
Same general question here. Any reason not to compare directly to the `sklearn` classes?